app/services/submit_application.rb

```diff
@@ -31,5 +31,10 @@ def submit_application
       application_choice.edit_by = ApplicationDates.new(application_form).edit_by
       ApplicationStateChange.new(application_choice).submit!
     end
+
+    application_form.references.includes(:application_form).each do |ref|
```

Comment:
Perhaps we should do this in the same loop as the mail is sent, because the ordering is important.

Reply:
Yes, it's nice to have the update at the same time the email is sent.

I've just one small concern: `send_reference_request_email_to_referees` will do 2 different things, instead of just one. (but I think it's fine moving the update_token code in the model, have a look)